sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/MemorySinkV2Suite.scala

```diff
@@ -44,16 +46,16 @@ class MemorySinkV2Suite extends StreamTest with BeforeAndAfter {
     val writer = new MemoryStreamWriter(sink, OutputMode.Append(), new StructType().add("i", "int"))
     writer.commit(0,
       Array(
-        MemoryWriterCommitMessage(0, Seq(InternalRow(1), InternalRow(2))),
-        MemoryWriterCommitMessage(1, Seq(InternalRow(3), InternalRow(4))),
-        MemoryWriterCommitMessage(2, Seq(InternalRow(6), InternalRow(7)))
+        MemoryWriterCommitMessage(0, Seq(Row(1), Row(2))),
```

Comment:
Why was this changed back to Row?

Reply:
Now the `DataWriter` needs to copy the input row before buffering it, which can be done by the `RowEncoder` when converting `InternalRow` to `Row`. Then the write message carries `Row`s to the driver side.

Reply:
Why not use InternalRow.copy? I'd rather keep the update to InternalRow, but as long as the tests pass I wouldn't block this.

Reply:
because the memory sink needs `Row`s at the end. Instead of collecting `InternalRow`s via copy and then convert to `Row`s, I think it's more efficient to collect `Row`s directly.

Reply:
Sounds good. Thanks for explaining your rationale.